front/src/common/BootstrapSNCF/InputGroupSNCF.tsx

```diff
@@ -149,64 +140,74 @@ export default function InputGroupSNCF({
-        <div className={`input-group-${orientation === 'left' ? 'prepend' : 'append'}`}>
+        <div
+          className={cx({
+            'input-group-prepend': orientation === 'left',
+            'input-group-append': orientation !== 'left',
+          })}
+        >
           {' '}
           <div className="btn-group dropdown">
             <button
               type="button"
-              className={cx('btn btn-secondary dropdown-toggle', condensed && 'pr-2 pl-2')}
+              className={cx('btn btn-secondary dropdown-toggle', {
+                'pr-2 pl-2': condensed,
+              })}
               onClick={() => setIsDropdownShown(!isDropdownShown)}
               aria-haspopup="true"
               aria-expanded="false"
               aria-controls={id}
               disabled={disabled || disableUnitSelector}
             >
-              <span className={cx(condensed && 'small')}>{selected.label}</span>
+              <span className={cx({ small: condensed })}>{selectedUnit.label}</span>
               <i
-                className={cx(
-                  isDropdownShown ? 'icons-arrow-up' : 'icons-arrow-down',
-                  condensed && ' ml-2'
-                )}
+                className={cx(isDropdownShown ? 'icons-arrow-up' : 'icons-arrow-down', {
+                  'ml-2': condensed,
+                })}
                 aria-hidden="true"
               />
             </button>
             <div
-              className={cx(
-                'dropdown-menu dropdown-menu-right osrd-dropdown-sncf',
-                isDropdownShown && 'show'
-              )}
+              className={cx('dropdown-menu dropdown-menu-right osrd-dropdown-sncf', {
+                show: isDropdownShown,
+              })}
               id={id}
               // eslint-disable-next-line react/no-unknown-property
               x-placement="bottom-end"
             >
-              <ul>
-                {options.map((option) => (
-                  <li key={nextId()}>
-                    <label htmlFor={option.id} className="dropdown-item">
-                      <div
-                        onClick={() => {
-                          setSelected(option);
-                          handleType({ type: option.id, value: 0 });
-                          setIsDropdownShown(false);
-                        }}
-                        role="button"
-                        tabIndex={0}
-                      >
-                        {option.label}
-                      </div>
-                    </label>
-                    <input
-                      type="radio"
-                      name={id}
-                      value={option.id}
-                      id={option.id}
-                      className="sr-only"
-                    />
-                  </li>
-                ))}
-              </ul>
+              {options.map((option) => (
+                <div
+                  key={nextId()}
```

Comment:
Don't use nextId, rather use the id of the option
```suggestion
                  key={option.id}
```

Reply:
Done